### PR TITLE
feat: enable custom theme to be provided to iframe embed

### PR DESCRIPTION
**Pull request by venarius**
- **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
This adds the ability to provide a custom theme for the iFrame embed

  I have also written out the complete options for the theme into the docs so it's easier for users to understand which options are possible

- **What is the new behavior (if this is a feature change)?**
fixes https://github.com/novuhq/novu/issues/1041
Screenshot 2022-08-17 033429
Screenshot 2022-08-17 033442
Screenshot 2022-08-17 040442

### CODE REVIEW

File: libs/embed/src/embed.ts [hunk 1]
```diff
@@ -17,6 +17,8 @@ class Novu {
 
   private socketUrl?: string = '';
 
+  private theme?: Record<string, unknown>;
```
Comment (davidsoderberg) [inline libs/embed/src/embed.ts:20]: It would be great if we could create an interface or type for the theme 😄

**Review by davidsoderberg — APPROVED** [review]
Looks good just had one small comment 🎉

File: libs/embed/src/embed.ts (hunk 1 above)
Comment (venarius) [reply libs/embed/src/embed.ts]: Sure! I thought of the same but I personally saw no advantage of doing so as this only bloats the devDependencies of the embed lib and in the end the user only receives a JS file with no types anyway 😅

File: apps/widget/src/components/notification-center/NotificationCenterWidget.tsx [hunk 2]
```diff
@@ -1,6 +1,7 @@
 import { NotificationCenter, NovuProvider } from '@novu/notification-center';
+import { INovuThemeProvider } from '@novu/notification-center/dist/esm/types/store/novu-theme-provider.context';
```
Comment (scopsy) [inline apps/widget/src/components/notification-center/NotificationCenterWidget.tsx:2]: @venarius I think you can use `INovuTheme` from root component which is exported by the notification center:
```suggestion
import { INovuTheme } from '@novu/notification-center';
```

File: libs/embed/src/embed.ts [hunk 3]
```diff
@@ -6,6 +6,7 @@ import iFrameResize from 'iframe-resizer';
 import * as EventTypes from './shared/eventTypes';
 import { UnmountedError, DomainVerificationError } from './shared/errors';
 import { IFRAME_URL } from './shared/resources';
+import { INovuThemeProvider } from '@novu/notification-center/dist/esm/types/store/novu-theme-provider.context';
```
Comment (scopsy) [inline libs/embed/src/embed.ts:9]:
```suggestion
import { INovuTheme } from '@novu/notification-center';
```

the same can be applied here, the theme provider is an internal interface so better to use this one 🥳

File: pnpm-lock.yaml [hunk 4]
```diff
@@ -1,4 +1,4 @@
-lockfileVersion: 5.4
+lockfileVersion: 5.3
```
Comment (scopsy) [inline pnpm-lock.yaml:1]: Could you check your pnpm version? looks like you are using an older one. 7+ can be safely used

File: libs/embed/src/embed.ts [hunk 5]
```diff
@@ -60,6 +63,7 @@ class Novu {
       this.options = selectorOrOptions;
       this.backendUrl = selectorOrOptions.backendUrl;
       this.socketUrl = selectorOrOptions.socketUrl;
+      this.theme = selectorOrOptions.theme;
```
Comment (scopsy) [inline libs/embed/src/embed.ts:66]: The API is great here!

File: docs/docs/notification-center/iframe-embed.md [hunk 6]
```diff
@@ -53,7 +53,100 @@ Optionally the embed init script receives a position object, you can use it to s
 </script>
 ```
 
-### Enabling HMAC Encryption
+## Customizing the theme
+
+The notification center component can be customized by passing a `theme` to the init script.
+
+```html
```
Comment (scopsy) [inline docs/docs/notification-center/iframe-embed.md:60]: I think here we can point to https://docs.novu.co/notification-center/react-components#customizing-the-notification-center-theme

So we have a nice table view of all properties in a single place. And here I would remove the implementation of the light dark properties. Wdyt?

File: pnpm-lock.yaml (hunk 4 above)
Comment (venarius) [reply pnpm-lock.yaml:1]: Hmm, I installed them using ```npm run setup:project``` which runs ```npx pnpm@7.3.0 i && node scripts/setup-env-files.js && npx pnpm build```

So every dev should be using pnpm at version 7.3.0 when running this script, or?

File: libs/embed/src/embed.ts (hunk 3 above)
Comment (venarius) [reply libs/embed/src/embed.ts:9]: See my answer above 😅

File: docs/docs/notification-center/iframe-embed.md (hunk 6 above)
Comment (venarius) [reply docs/docs/notification-center/iframe-embed.md:60]: Yeah, that's a better idea! For the other question - I answered it above with my point of view :)

**Comment by scopsy** [comment]
@venarius I've merged the 0.7.0 and GitHub automatically closed this pr :(( Sorry about that! could you please reopen it against main?

**Pull request closed without merging**